### PR TITLE
Perform statistics printing via the API

### PR DESCRIPTION
**Pull request by nafur**
This PR changes how the command executor prints the statistics by moving stuff around a bit, eventually using only proper API methods of `api::Solver`. This PR also removes the `smt_engine.h` include from `command_executor.cpp`.

### CODE REVIEW

File: src/api/cpp/cvc5.cpp [hunk 1]
```diff
@@ -6656,7 +6664,9 @@ Result Solver::checkEntailed(const std::vector<Term>& terms) const
          "(try --incremental)";
   CVC5_API_SOLVER_CHECK_TERMS(terms);
   //////// all checks before this line
-  return d_smtEngine->checkEntailed(Term::termVectorToNodes(terms));
+  cvc5::Result r = d_smtEngine->checkEntailed(Term::termVectorToNodes(terms));
+  d_smtEngine->printStatisticsDiff();
```
Comment (ajreynol) [inline src/api/cpp/cvc5.cpp:6668]: Can we actually push this further into SmtEngine::checkEntailed / SmtEngine::checkSat?

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: src/api/cpp/cvc5.cpp (hunk 1 above)
Comment (nafur) [reply src/api/cpp/cvc5.cpp]: I've put it into `checkSatInternal`, so it only lives in a single place. (well, actually two when an exception is thrown...)

File: src/smt/smt_engine.cpp [hunk 2]
```diff
@@ -1925,10 +1927,13 @@ void SmtEngine::printStatisticsSafe(int fd) const
   d_env->getStatisticsRegistry().printSafe(fd);
 }
 
-void SmtEngine::printStatisticsDiff(std::ostream& out) const
+void SmtEngine::printStatisticsDiff() const
 {
-  d_env->getStatisticsRegistry().printDiff(out);
-  d_env->getStatisticsRegistry().storeSnapshot();
+  if (d_env->getOptions().base.statisticsEveryQuery)
```
Comment (ajreynol) [inline src/smt/smt_engine.cpp:1932]: Maybe move this to guard the call to this method, not guard within the method itself.

**Review by ajreynol — CHANGES_REQUESTED** [review]
One minor request, otherwise looks good

**Review by ajreynol — APPROVED** [review]

**Pull request merged**